pkg/armhelpers/util.go

```diff
@@ -35,18 +37,60 @@ func SplitBlobURI(URI string) (string, string, string, error) {
 	return accountName, containerName, blobPath, nil
 }
 
-// ByVMNameOffset implements sort.Interface for []VirtualMachine based on
-// the Name field.
-type ByVMNameOffset []compute.VirtualMachine
+// LinuxVMNameParts returns parts of Linux VM name e.g: k8s-agentpool1-11290731-0
+func LinuxVMNameParts(vmName string) (orchestrator string, poolIdentifier string, nameSuffix string, agentIndex int, err error) {
+	vmNameParts := strings.Split(vmName, "-")
 
-func (vm ByVMNameOffset) Len() int      { return len(vm) }
-func (vm ByVMNameOffset) Swap(i, j int) { vm[i], vm[j] = vm[j], vm[i] }
-func (vm ByVMNameOffset) Less(i, j int) bool {
-	vm1NameParts := strings.Split(*vm[i].Name, "-")
-	vm1Num := vm1NameParts[len(vm1NameParts)-1]
+	if len(vmNameParts) != 4 {
+		return "", "", "", -1, fmt.Errorf("resource name was missing from identifier")
+	}
+
+	vmNum, err := strconv.Atoi(vmNameParts[3])
+
+	if err != nil {
+		return "", "", "", -1, fmt.Errorf("Error parsing VM Name: %v", err)
+	}
+
+	return vmNameParts[0], vmNameParts[1], vmNameParts[2], vmNum, nil
+}
+
+// WindowsVMNameParts returns parts of Windows VM name e.g: 50621acs9000
+func WindowsVMNameParts(vmName string) (poolPrefix string, acsStr string, poolIndex int, agentIndex int, err error) {
+	poolPrefix = strings.Split(vmName, "acs")[0]
+	poolInfo := strings.Split(vmName, "acs")[1]
+
+	poolIndex, err = strconv.Atoi(poolInfo[:3])
+	if err != nil {
+		return "", "", -1, -1, fmt.Errorf("Error parsing VM Name: %v", err)
+	}
+
+	agentIndex, _ = strconv.Atoi(poolInfo[3:])
+	fmt.Printf("%d\n", agentIndex)
 
-	vm2NameParts := strings.Split(*vm[j].Name, "-")
-	vm2Num := vm2NameParts[len(vm2NameParts)-1]
+	if err != nil {
+		return "", "", -1, -1, fmt.Errorf("Error parsing VM Name: %v", err)
+	}
+
+	return poolPrefix, "acs", poolIndex, agentIndex, nil
+}
+
+// GetVMNameIndex return VM index of a node in the Kubernetes cluster
+func GetVMNameIndex(osType compute.OperatingSystemTypes, vmName string) (int, error) {
```

Comment:
nit, feel free to ignore: Why not make this take the VM properties blob? Then it could reach in and look at the OS Disk OSType. No need for the caller to pull it out.